### ISSUE TITLE
Doesn't handle malformed aws.availability_zone labels

### ISSUE THREAD

**Issue by schuylr** [ampedandwired/rancher-reaper#2]
Great PoC! Nicely documented and easy to get running.

I think you need to either derive the Region from the AWS Availability Zone or pass the region as an entirely separate label. Either way, the AWS CLI (and SDKs respectively) need to use the region only. You could `gsub` out the last letter from the AZ name, maybe.

```
10/25/2016 10:26:26 PMI, [2016-10-26T02:26:26.509312 #5]  INFO -- : Reaping terminated AWS hosts...
10/25/2016 10:26:26 PMW, [2016-10-26T02:26:26.509420 #5]  WARN -- : *** Dry run - no changes will be applied
10/25/2016 10:26:26 PME, [2016-10-26T02:26:26.856604 #5] ERROR -- : :region option must a region name, not an availability zone name; try `eu-central-1' instead of `eu-central-1a' (ArgumentError)
10/25/2016 10:26:26 PM/usr/local/bundle/gems/aws-sdk-core-2.6.3/lib/aws-sdk-core/plugins/ec2_region_validation.rb:11:in `after_initialize'
10/25/2016 10:26:26 PM/usr/local/bundle/gems/aws-sdk-core-2.6.3/lib/seahorse/client/base.rb:84:in `block in after_initialize'
10/25/2016 10:26:26 PM/usr/local/bundle/gems/aws-sdk-core-2.6.3/lib/seahorse/client/base.rb:83:in `each'
10/25/2016 10:26:26 PM/usr/local/bundle/gems/aws-sdk-core-2.6.3/lib/seahorse/client/base.rb:83:in `after_initialize'
10/25/2016 10:26:26 PM/usr/local/bundle/gems/aws-sdk-core-2.6.3/lib/seahorse/client/base.rb:21:in `initialize'
10/25/2016 10:26:26 PM/usr/local/bundle/gems/aws-sdk-core-2.6.3/lib/seahorse/client/base.rb:105:in `new'
10/25/2016 10:26:26 PM/usr/src/app/reaper.rb:66:in `host_terminated?'
10/25/2016 10:26:26 PM/usr/src/app/reaper.rb:38:in `block in reap_hosts'
10/25/2016 10:26:26 PM/usr/src/app/rancher_api.rb:20:in `yield'
10/25/2016 10:26:26 PM/usr/src/app/rancher_api.rb:20:in `block (3 levels) in get_all'
10/25/2016 10:26:26 PM/usr/src/app/rancher_api.rb:20:in `each'
10/25/2016 10:26:26 PM/usr/src/app/rancher_api.rb:20:in `block (2 levels) in get_all'
10/25/2016 10:26:26 PM/usr/src/app/rancher_api.rb:17:in `loop'
10/25/2016 10:26:26 PM/usr/src/app/rancher_api.rb:17:in `block in get_all'
10/25/2016 10:26:26 PM/usr/src/app/reaper.rb:37:in `each'
10/25/2016 10:26:26 PM/usr/src/app/reaper.rb:37:in `each'
10/25/2016 10:26:26 PM/usr/src/app/reaper.rb:37:in `reap_hosts'
10/25/2016 10:26:26 PM/usr/src/app/reaper.rb:21:in `run'
10/25/2016 10:26:26 PMconfig.ru:24:in `block (2 levels) in <main>'
```

**Comment by schuylr** [comment]
Checked your code, I figured it out. I screwed up my `aws.availability_zone` host label when I was trying to interpolate it from the AWS metadata, and thus it turned `eu-central-1a'` into `eu-central-1a`. See:

![screen shot 2016-10-25 at 10 33 35 pm](https://cloud.githubusercontent.com/assets/12628715/19711366/4039193c-9b03-11e6-91e2-940326335893.png)

In light of this, I'd say there should just be another rescue to catch a malformed `aws.availability_zone` label. Hope this helps!
